types/constants.ts

```diff
@@ -1 +1,62 @@
-export const ARNS_NAME_REGEX = new RegExp('^(?!-)[a-z0-9-s+]{1,32}(?<!-)$');
+export const ARNS_NAME_REGEX = new RegExp('^(?!-)[a-zA-Z0-9-s+]{1,32}(?<!-)$');
+
+export const TIER_DATA: { [x: number]: string[] } = {
+  1: [
+    'Up to 100 Subdomains',
+    'Available via all ArNS-enabled gateways',
+    'Permanently stored on Arweave',
+  ],
+  2: [
+    'Up to 1,000 Subdomains',
+    'Available via all ArNS-enabled gateways',
+    'Permanently stored on Arweave',
+  ],
+  3: [
+    'Up to 10,000 Subdomains',
+    'Available via all ArNS-enabled gateways',
+    'Permanently stored on Arweave',
+  ],
+};
+export const NAME_PRICE_INFO =
+  'Registration fees are determined by the character length of the domain, lease duration, and what tier you choose.';
+
+export function NAME_PRICE_CALC({
+  domain,
+  years,
+  tier,
+  fees,
+}: {
+  domain: string | undefined;
```

Comment:
whenever you want to optionally define a param, you can use `?` instead of `| undefined`. 

```suggestion
  domain?: string;
```